lib/pool.dart

```diff
@@ -191,17 +191,21 @@ class Pool {
       }
     }
 
-    Future? doneFuture;
+    Future<void>? doneFuture;
 
     void onListen() {
       iterator = elements.iterator;
 
       assert(doneFuture == null);
-      doneFuture = Future.wait(
-              Iterable<int>.generate(_maxAllocatedResources)
-                  .map((i) => withResource(() => run(i))),
-              eagerError: true)
-          .catchError(controller.addError);
+      var futures = Iterable<int>.generate(_maxAllocatedResources)
+          .map((i) => withResource(() => run(i)));
+      doneFuture = Future.wait(futures, eagerError: true)
+          .catchError((Object e, StackTrace s) {
+        controller.addError(e, s);
+        // We now have to return a valid value from catchError or else you get
```

Comment:
Another option would be to translate into a `Future<void>` first.

```dart
doneFuture = Future.wait(future, eagerError: true)
    .then<void>((_) {})
    .catchError(controller.addError);
```

Reply:
that is probably a bit cleaner, ill change it to that